### ISSUE TITLE
bug(cdk/overlay): attach() and detach are not working in zoneless apps

### ISSUE THREAD

**Issue by javiermarinros** [angular/components#28456]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

The Overlay `attach()` and `detach()` methods are relying on the `ngZone.onStable` observable that is never invoked using the new `provideZonelessChangeDetection` app provider.

The related actual code:

```
  // Update the position once the zone is stable so that the overlay will be fully rendered
    // before attempting to position it, as the position may depend on the size of the rendered
    // content.
    this._ngZone.onStable.pipe(take(1)).subscribe(() => {
      // The overlay could've been detached before the zone has stabilized.
      if (this.hasAttached()) {
        this.updatePosition();
      }
    });
```

### Expected Behavior

Overlays to attach and position themselves in zoneless aplications

### Actual Behavior

Overlay is created but not positioned and not visible

### Environment

- Angular: 17.1
- CDK/Material: 17.1

**Comment by atscott** [comment]
Indeed, this is the case everywhere in the repo that uses NgZone APIs. This particular one has a WIP fix here https://github.com/angular/components/pull/28332

**Comment by crisbeto** [comment]
Should be fixed now that #28332 has landed.
